src/components/TokenBalance.tsx

```diff
@@ -87,20 +92,21 @@ function TokenBalance({
         <Image source={{ uri: tokensWithUsdValue[0].imageUrl }} style={styles.tokenImg} />
         <View style={styles.column}>
           <Text style={style} testID={'TotalTokenBalance'}>
-            {localCurrencySymbol}
-            {totalTokenBalanceLocal?.toFormat(2) ?? '-'}
+            {!hideBalance && localCurrencySymbol}
+            {hideBalance ? 'XX.XX' : totalTokenBalanceLocal?.toFormat(2) ?? '-'}
           </Text>
           <Text style={styles.tokenBalance}>
-            {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
+            {!hideBalance && formatValueToDisplay(tokenBalance)}{' '}
+            {!hideBalance && tokensWithUsdValue[0].symbol}
           </Text>
```

Comment:
nit: you can just skip rendering this Text component if hideBalances is true right?
```
{!hideBalance &&
(<Text style={styles.tokenBalance}>
  {formatValueToDisplay(tokenBalance)} {tokensWithUsdValue[0].symbol}
</Text>)
}
```